pkg/rpm/salt.spec

```diff
@@ -456,7 +472,11 @@ ln -s -f /opt/saltstack/salt/salt-cloud %{_bindir}/salt-cloud
 
 
 %post master
-%systemd_post salt-master.service
+# %systemd_post salt-master.service
+if [ $1 -eq 1 ] ; then
```

Comment:
-eq 1 is an install, but you are neglecting upgrade that is '$1` == 2' where you would want to restart the salt-xxx if it was already running.

Do you need an 
if [ $1 -gt 1 ] ; then
  systemctl retry-restart salt-xxxx.service > /dev/null 2>&1 || :+1: 
else
 systemctl preset salt-xxxx.service > /dev/null 2>&1 || :
fi